src/transformers/models/maskformer/modeling_maskformer.py

```diff
@@ -1424,7 +1424,11 @@ def forward(
         # repeat the queries "q c -> b q c"
         batch_size = image_features.shape[0]
         queries_embeddings = self.queries_embedder.weight.unsqueeze(0).repeat(batch_size, 1, 1)
-        inputs_embeds = torch.zeros_like(queries_embeddings, requires_grad=True)
+        inputs_embeds = torch.zeros_like(queries_embeddings)
+
+        # torch.export.export does no support requires_grad
+        if self.training:
+            inputs_embeds.requires_grad_(True)
```

Comment:
just curious if torch.export should/could be run in torch.no_grad without this change

Reply:
No, it does not work that way :) here is min reproducing example:

```python
import torch

class Module(torch.nn.Module):
    def __init__(self):
        super().__init__()
        self.linear = torch.nn.Linear(10, 10)

    def forward(self, x):
        b = torch.ones(10, requires_grad=True)
        return self.linear(x) + b

model = Module()

with torch.no_grad():
    torch.export.export(model, args=(torch.randn(1, 10),))
```